pkg/compact/downsample/downsample.go

```diff
@@ -100,7 +102,7 @@ func Downsample(
 
 	var (
 		aggrChunks []*AggrChunk
-		all        []sample
+		all        []tsdbutil.Sample
```

Comment:
So in theory this struct was supposed to be used only for testing. It's controversial, but the idea is that we wanted to have reproduced sample struct everywhere as it's easy to copy such small struct and easier to type it (as you could see when refactoring).

Indeed it makes it's difficult to reuse though... I am not sure if we want to use Prometheus struct designed for testing though in our production code 🤔

Reply:
Got it. I see. Then what about exposing `downsample.sample` struct and use it?

Reply:
Updated with the way mentioned in the OH.